### PR TITLE
NUT-6097: add bar slider chart

### PR DESCRIPTION
**Pull request by mikeathan-nhs**
### Developer checklist for this pull request

- [x] Documentation has been updated (README's, CHANGELOG etc.)
- [x] Software builds without errors or warnings
- [x] Any debugger/console.log() statements and commented out code have been removed
- [x] Unit test suites have been run, all tests are passing and absent of console warnings/errors
- [x] Make sure that new code is unit tested and linted.

### Description

Add generic chart component

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,10 +2,12 @@
 
 This repository contains the code for NHS.UK Chart components
 
-These chart components are designed for visualizing Body Mass Index (BMI), Child Body Mass Index (CBMI), Blood Pressure (BP) and Waist to Height (WHTR) data.
+These chart components are designed for visualizing Body Mass Index (BMI), Child Body Mass Index (CBMI) and Blood Pressure (BP) data.
```
Comment (JoshuaBates-NHS) [inline README.md:5]: It's worth rewriting this section and call out the components which this refers to

File: src/models/bar-slider-chart.model.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+      return {
+        label: tick.value.toString(),
+        value: this.calculateRawPosition(tick.value),
+        isVisible,
+      };
+    });
+  }
+
+  ChartRegions(): BarSliderRegionPosition[] {
```
Comment (JoshuaBates-NHS) [inline src/models/bar-slider-chart.model.ts:46]: Should these methods be upper case?

File: src/custom-types/bar-slider.types.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+export type BarSliderConstraints = {
+  max: {
+    value: number;
+    position: number;
+  };
+};
+
+export type BarSlidertAxis = {
```
Comment (JoshuaBates-NHS) [inline src/custom-types/bar-slider.types.ts:15]: should this have a t before Axis?

File: src/components/styles/mixins.module.scss [hunk 4]
```diff
@@ -49,7 +49,7 @@ $marker-axis-offset: 5px;
   bottom: -5px;
   margin-left: -0.5px;
   box-shadow: 0px 0px white inset, 1px 0px white, -1px 0px white;
-  z-index: 4;
+  z-index: 10;
```
Comment (JoshuaBates-NHS) [inline src/components/styles/mixins.module.scss:52]: I'd throw an extra zero on this - we don't know how people might use their z-index but 10 isn't that many layers to play with

File: src/components/bar-slider/bar-slider-figure-chart.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  config,
+  value,
+  ariaLabel,
+  markerText,
+}: Readonly<BarSliderChartFigureProps>): JSX.Element {
+  const barSliderLegendKeys = Object.values(config.segments)
+    .filter((s) => s.legend != null)
+    .map((segment) => ({
+      text: segment.legend!,
+      className: `${config.name}-legend-key-${segment.stylePrefix}`,
```
Comment (JoshuaBates-NHS) [inline src/components/bar-slider/bar-slider-figure-chart.tsx:43]: called stylePrefix but it's being used as a suffix

**Comment by JoshuaBates-NHS** [comment]
discussed offline - sonarcloud analysis hasn't run, needs to be checked

File: README.md (hunk 1 above)
Comment (mikeathan-nhs) [reply README.md:5]: resolved

**Pull request merged**